src/core/structure/sheet.ts

```diff
@@ -377,7 +420,7 @@ export default class Sheet {
     const cell = new Cell();
     cell.formula = value;
```

Comment:
It is a bit not intuitive that we set the formula.
Can we have a function for setting the value and encapsulate the cell values instead.

I wish that we also force the user to call `revolveCell` somehow.

Like maybe we can have a function `equeueValue`, which takes the new value, then a function called `fulfillValue` which take a `ResolveCellResult` as parameter and updates the references. We probably need another function for updating `referredCell` references, but I'm all with that. 

I just don't like how we currently crossing the abstraction line between cell and sheet. Almost like cell is not an object in the oop sense, but then you applied encapsulation in the `setState`. So I think it should be either a data record without encapsulation or make it into a real oop object and give it ownership of its state by only modifying it with functions.

Reply:
One approach we could take instead is move responsibility for the `Cell` fields to `ExpressionHandler` or some similar static, "all sheets"-scope structure. This way we could move the logic out from `Sheet` and maybe improve the clarity a bit. I don't think solving this issue is within the scope of this PR, since it's just adding reference tracking to the existing structures.

Reply:
I don't think calling the variable `formula` is the right thing, I would prefer if revert it back to `value`, because it is a value, but it could be a formula, however it is not always a formula. I would be confused on how to set the value if the function is accepting a formula.

I'm fine with it as `value` if you don't want to make it into a function.

Reply:
No, the formula always corresponds to user input and the value is always the resolved value that's displayed in the cell in the UI. So semantically, the function accepting `value` doesn't make sense since we're always setting the formula.

Reply:
We still have the value stored 2 times, in `value` and in `formula` do you think that this is fine?

Reply:
I think it's fine for now. I think it'd be premature optimization at this point to include a mechanism to use either the formula or value depending on if the formula is an expression.